Source/Core/Core/IOS/IOSC.cpp

```diff
@@ -164,6 +167,54 @@ ReturnCode IOSC::Decrypt(Handle key_handle, u8* iv, const u8* input, size_t size
   return DecryptEncrypt(Common::AES::Mode::Decrypt, key_handle, iv, input, size, output, pid);
 }
 
+ReturnCode IOSC::VerifyPublicKeySign(const std::array<u8, 20>& sha1, Handle signer_handle,
+                                     const u8* signature, u32 pid) const
+{
+  if (!HasOwnership(signer_handle, pid))
+    return IOSC_EACCES;
+
+  const KeyEntry* entry = FindEntry(signer_handle, SearchMode::IncludeRootKey);
+  if (!entry)
+    return IOSC_EINVAL;
+
+  // TODO: add support for keypair entries.
+  if (entry->type != TYPE_PUBLIC_KEY)
+    return IOSC_INVALID_OBJTYPE;
+
+  switch (entry->subtype)
+  {
+  case SUBTYPE_RSA2048:
+  case SUBTYPE_RSA4096:
+  {
+    const size_t expected_key_size = entry->subtype == SUBTYPE_RSA2048 ? 0x100 : 0x200;
+    _assert_(entry->data.size() == expected_key_size);
+
+    mbedtls_rsa_context rsa;
+    mbedtls_rsa_init(&rsa, MBEDTLS_RSA_PKCS_V15, 0);
+    Common::ScopeGuard context_guard{[&rsa] { mbedtls_rsa_free(&rsa); }};
+
+    mbedtls_mpi_read_binary(&rsa.N, entry->data.data(), entry->data.size());
+    mbedtls_mpi_read_binary(&rsa.E, entry->misc_data.data(), entry->misc_data.size());
+    rsa.len = (mbedtls_mpi_bitlen(&rsa.N) + 7) >> 3;
```

Comment:
Oh, absolutely. I took this from [rsa_verify.c](https://tls.mbed.org/api/rsa__verify_8c_source.html) and forgot to change it. I'll fix this.